src/bos/server/redis_db_utils.py

```diff
@@ -106,6 +108,34 @@ def get_all(self):
             data.append(single_data)
         return data
 
+    def get_all_filtered(self,
+                         filter_func: Callable[[dict], dict | None],
+                         start_after_key: Optional[str] = None,
+                         page_size: int = 0) -> list[dict]:
+        """
+        Get an array of data for all keys after passing them through the specified filter
+        (discarding any for which the filter returns None)
+        If start_after_id is specified, all ids lexically <= that id will be skipped.
+        If page_size is specified, the list will be returned if it contains that many
+        elements, even if there may be more remaining.
+        """
+        data = []
+        for value in self.iter_values(start_after_key):
+            filtered_value = filter_func(value)
+            if filtered_value is not None:
```

Comment:
if filtered_value

Reply:
Although I don't think it is going to be the case, in theory the filter could produce an empty dictionary, which would mean we want to include it in our list. None is the specific value that indicates the filter was not matched. And as with my previous note, I believe this check is faster, and this check is one which is being made many times per request. Normally I don't worry that much about the efficiency of Python calls, but because these changes were specifically being made to reduce the time our large scale requests were taking, it made sense to me to try and be more optimal.